vortex-ffi/src/file.rs

```diff
@@ -33,6 +35,14 @@ pub struct vx_file_reader {
     pub(crate) inner: VortexFile,
 }
 
+/// A reference to a array stream being written to an external system (e.g. a file).
+/// Must be closed with `vx_array_stream_writer_close`.
+#[allow(non_camel_case_types)]
+pub struct vx_array_stream_writer {
+    // To support other writes use an enum of writers here.
+    writer: JoinHandle<VortexResult<File>>,
```

Comment:
This seems weird to me? We have partially opened the file?

Reply:
Its a open file which batches push into a sink are written too